react-app/src/actions/leagues.js

```diff
@@ -51,4 +54,36 @@ function shouldFetchLeague(state, leagueId) {
   } else {
     return league.didInvalidate
   }
-}
\ No newline at end of file
+}
+
+const buildLeagueSchedule = (league) => {
+
+  if (typeof (league) === 'undefined' || league == null) {
+    return [];
+  }
+
+  var scheduleWeeks = [];
+  let prevWeek = '';
+
+  league.scheduledMatches.forEach((match, i) => {
```

Comment:
league.scheduledMatches should be sorted before the .forEach is called and .forEach can probably be replaced with a .map

Reply:
Sorted by what?

Reply:
Date.weekNumber would make the most sense.. I'm just thinking if the list of scheduled matches is out of order for whatever reason (shouldn't happen but could) then a week could theoretically be rendered multiple times (since you're just creating new weeks whenever the weekNumber changes).

Really what would make sense is if you mapped the list of scheduled matches to a list of dateIds, sorted that by week number, then traversed that list and created your week objects. That way you don't even need the prevWeek stuff.

Reply:
just sort, don't fuck around with reducing it's not worth your time

Reply:
.sort((scheduleMatchOne, scheduleMatchTwo) => {
let dateOne = LeagueHelpers.getDate(league, scheduleMatchOne.dateId);
let dateOne = LeagueHelpers.getDate(league, scheduleMatchTwo.dateId);
return dateOne.weekNumber - dateTwo.weekNumber
});